src/vm/eventpipe.cpp

```diff
@@ -525,30 +569,66 @@ void WINAPI EventPipe::SwitchToNextFileTimerCallback(PVOID parameter, BOOLEAN ti
     // Take the lock control lock to make sure that tracing isn't disabled during this operation.
     CrstHolder _crst(GetLock());
 
+    if (s_pSession == nullptr || s_pFile == nullptr)
+        return;
+
     // Make sure that we should actually switch files.
-    UINT64 multiFileTraceLengthInSeconds = s_pSession->GetMultiFileTraceLengthInSeconds();
-    if (!Enabled() || s_pSession->GetSessionType() != EventPipeSessionType::File || multiFileTraceLengthInSeconds == 0)
-    {
+    if (!Enabled() || s_pSession->GetSessionType() != EventPipeSessionType::File || s_multiFileTraceLengthInSeconds == 0)
         return;
-    }
 
     GCX_PREEMP();
 
-    if (CLRGetTickCount64() > (s_lastFileSwitchTime + (multiFileTraceLengthInSeconds * 1000)))
+    if (CLRGetTickCount64() > (s_lastFileSwitchTime + (s_multiFileTraceLengthInSeconds * 1000)))
     {
         SwitchToNextFile();
         s_lastFileSwitchTime = CLRGetTickCount64();
     }
 }
 
+void WINAPI EventPipe::FlushTimer(PVOID parameter, BOOLEAN timerFired)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
+        PRECONDITION(timerFired);
+    }
+    CONTRACTL_END;
+
+    // Take the lock control lock to make sure that tracing isn't disabled during this operation.
+    CrstHolder _crst(GetLock());
+
+    if (s_pSession == nullptr || s_pFile == nullptr)
+        return;
+
+    // Make sure that we should actually switch files.
+    if (!Enabled() || s_pSession->GetSessionType() != EventPipeSessionType::IpcStream)
+        return;
+
+    GCX_PREEMP();
+
+    if (CLRGetTickCount64() > (s_lastFileSwitchTime + 100))
+    {
+        // Get the current time stamp.
+        // WriteAllBuffersToFile will use this to ensure that no events after
+        // the current timestamp are written into the file.
+        LARGE_INTEGER stopTimeStamp;
+        QueryPerformanceCounter(&stopTimeStamp);
+        s_pBufferManager->WriteAllBuffersToFile(s_pFile, stopTimeStamp);
+
+        s_lastFileSwitchTime = CLRGetTickCount64();
```

Comment:
Correct me if I'm wrong, but this timestamp is essentially the timestamp for "last time we flushed buffer via IPC", right? Is there any chance we can rename this to something more representative of what this is, rather than `s_lastFileSwitchTime` since that makes it sound like we're in multi-file switch mode?